### DISCUSSION TITLE
With takeover mode, how to format on save plain typescript files?

### DISCUSSION THREAD

**Discussion by vidal7** [vuejs/language-tools#1908] [Q&A]
Hi there,

First at all, thank you for making this great extension.

We have a big codebase here and we are slowly switching from Vue2/No SFC/vue-class-component API to Vue3/SFC/composition API and I think this extension will enhanced the DX (Developer Experience) of our team.

Since it is recommended to activate the take over mode and since we still have many plain ts files, our formatter for plain ts files "vscode.typescript-language-features" is not available anymore.

Here's is our settings.json configuration.

```json
{
    "editor.formatOnSave": true,
    "[typescript]": {
        "editor.defaultFormatter": "vscode.typescript-language-features"
    },
    "[html]": {
        "editor.defaultFormatter": "vscode.html-language-features"
    },
    "[vue]": {
        "editor.defaultFormatter": "Vue.volar"
    }
}
```

So how format on save plain ts files? I know that we can use another formatter like prettier, but for now we prefer not to use it. We are happy with the vscode.typescript-language-features formatter. Searching on the web and as far as I know, vs code typescript default formatter can't be extracted or used without the builtin typescript extension that must be desactivated to use the take over mode.

**Comment by johnsoncodehk** [comment]
Volar uses the same formatter as `vscode.typescript-language-features` for TS documents, so you only need change `"editor.defaultFormatter": "vscode.typescript-language-features"` to `"editor.defaultFormatter": "Vue.volar"` in `"[typescript]"`.

**Reply by vidal7** [reply]
I tried it and it doesn't seems to work even after restarting vscode. I have the default formatter in script part of vue file working but not in plain .ts files. I am using the last version of volar. FYI, I am using Vue 2.7 and typescript 4.8.2.

**Reply by johnsoncodehk** [reply]
This may be the same issue as #1903.

**Reply by johnsoncodehk** [reply]
See https://github.com/johnsoncodehk/volar/issues/1903#issuecomment-1258549712

**Reply by vidal7** [reply]
WoW. That was quick! Can't wait to try it. For my information, how can we install a beta version of the extension to try it? Beta versions do not seem to be in the list when selecting others versions of it?

![image](https://user-images.githubusercontent.com/33458053/192371256-80a32fd7-7313-4103-a974-9709dd416a89.png)

**Reply by vidal7** [reply]
Ok, here is the way to do it https://github.com/johnsoncodehk/volar/issues/1880

**Reply by johnsoncodehk** [reply]
You need to download vsix and drag it to VSCode "Extensions" window: #1880

**Reply by vidal7** [reply]
Thank you. I can now confirm that v1.0.0-beta.0 has the right fix and formatting plain ts files now work again.
